--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,8 +105,9 @@
 PKG_CHECK_MODULES(CHEWING chewing>=${LIBCHEWING_VER})
 EXECUTE_PROCESS(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=datadir chewing
     OUTPUT_VARIABLE CHEWING_DATA_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
-SET_COMPILE_ENV(CHEWING_DATA_DIR "${CHEWING_DATA_DIR}" DISPLAY PATH
-    "Chewing data dir")
+add_definitions(-DCHEWING_DATA_DIR="${CHEWING_DATA_DIR}")
+#SET_COMPILE_ENV(CHEWING_DATA_DIR '"${CHEWING_DATA_DIR}"' DISPLAY PATH
+#    "Chewing data dir")
 
 FIND_PROGRAM(GOB2_CMD gob2)
 IF(${GOB2_CMD} STREQUAL "GOB2_CMD-NOTFOUND")
--- a/src/main.c
+++ b/src/main.c
@@ -163,7 +163,7 @@
     if (showFlags){
         printf("PROJECT_NAME=" quote_me(PROJECT_NAME) "\n");
         printf("DATA_DIR=" quote_me(DATA_DIR) "\n");
-        printf("CHEWING_DATA_DIR=" quote_me(CHEWING_DATA_DIR) "\n");
+        printf("CHEWING_DATA_DIR=" CHEWING_DATA_DIR "\n");
     }else{
         start_component ();
     }
--- a/src/IBusChewingEngine.gob
+++ b/src/IBusChewingEngine.gob
@@ -248,8 +248,8 @@
 	    gchar buf[100];
 	    g_snprintf(buf,100,"%s/.chewing",getenv("HOME"));
 
-	    G_DEBUG_MSG(6,"[I6] init() CHEWING_DATA_DIR=%s",  quote_me(CHEWING_DATA_DIR) );
-	    chewing_Init(quote_me(CHEWING_DATA_DIR), buf);
+	    G_DEBUG_MSG(6,"[I6] init() CHEWING_DATA_DIR=%s", CHEWING_DATA_DIR);
+	    chewing_Init(CHEWING_DATA_DIR, buf);
 
 	    self->context=chewing_new();
 	    chewing_set_ChiEngMode(self->context,CHINESE_MODE);
