Description: Fix Exec in desktop file
---
Forwarded: <no|not-needed|url proving that it has been forwarded>
--- ibus-chewing-1.4.14.orig/CMakeLists.txt
+++ ibus-chewing-1.4.14/CMakeLists.txt
@@ -229,6 +229,7 @@ IF(NOT GNOME_SHELL)
 ENDIF(NOT GNOME_SHELL)
 
 CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/data/chewing.xml.in ${CMAKE_BINARY_DIR}/data/chewing.xml)
+CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/data/ibus-setup-chewing.desktop.in ${CMAKE_BINARY_DIR}/data/ibus-setup-chewing.desktop)
 # Sub directories
 ADD_SUBDIRECTORY(src bin)
 ADD_SUBDIRECTORY(po)
@@ -245,7 +246,7 @@ INSTALL(FILES AUTHORS README ChangeLog C
 INSTALL(FILES ${CMAKE_BINARY_DIR}/data/chewing.xml 
     DESTINATION "${DATA_DIR}/ibus/component")
 
-INSTALL(FILES ${CMAKE_SOURCE_DIR}/data/ibus-setup-chewing.desktop 
+INSTALL(FILES ${CMAKE_BINARY_DIR}/data/ibus-setup-chewing.desktop 
     DESTINATION "${DATA_DIR}/applications")
 
 INSTALL(DIRECTORY icons
--- /dev/null
+++ ibus-chewing-1.4.14/data/ibus-setup-chewing.desktop.in
@@ -0,0 +1,12 @@
+[Desktop Entry]
+Name=IBus Chewing Preferences
+Name[zh_TW]=IBus 酷音偏好設定
+Comment=Set IBus Chewing Preferences
+Comment[zh_TW]=設定 IBus 酷音偏好
+Exec=@LIBEXEC_DIR/ibus-setup-chewing
+Icon=ibus-setup-chewing
+NoDisplay=true
+Terminal=false
+Type=Application
+StartupNotify=true
+Categories=Settings;
